### PR TITLE
fixes show all extensions button #10445  #10625

### LINKED ISSUE #10445: "Show All" button on Welcome page doesn't work
image

Clicking the link doesn't do anything

(note this is a regression from April release)

### LINKED ISSUE #10625: A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard:The show all button is not activated on pressing enter key.
[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:**#A11y_AzureDataStudioStayHealthy;#A11ySev1;#A11yTCS;#A11yMAS;#NonBenchmark#DesktopApp;#MAS2.1.1;#SQL Azure Data Studio;#FTP;#Keyboard;#STP;

**Environment Details:**
Application Name: Azure Data Studio
Version: 1.19.0-insider
Commit: e8dc0d1
Date: 2020-05-27T17:24:57.733Z
VS Code: 1.46.0
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

**Additional Details:**
MAS References:MAS2.1.1

**Repro Steps:**
1 . Launch Azure Data Studio Insiders application.
2 . If Welcome screen is not shown on application launch, then navigate to Help --> Welcome.
3 . Navigate Extend your data Studio on Welcome screen.
4 . Navigate to show all button using tab key.

**Actual:**
On navigating to show all button using tab key press enter the button does not get activated.

**Expected:**
There should be an action performed when enter is pressed on the button show all.

**User Impact:**
Keyboard users will not be able to access the information from activating the button.

**Attachment link for Reference:**
[10625A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard.zip](https://github.com/microsoft/azuredatastudio/files/4701226/10625A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard.zip)

### PR DESCRIPTION
**Pull request by v-bbrady**
This PR fixes the show all button on the Welcome Page by updating the command

This PR fixes #10445
This PR fixes #10625

### CODE REVIEW

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 1]
```diff
@@ -45,13 +45,13 @@ import { IHostService } from 'vs/workbench/services/host/browser/host';
 import { IProductService } from 'vs/platform/product/common/productService';
 import { KeyCode } from 'vs/base/common/keyCodes';
 import { joinPath } from 'vs/base/common/resources';
-import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+// import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:48]: remove

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 2]
```diff
@@ -471,9 +510,8 @@ class WelcomePage extends Disposable {
 	}
 
 	private createDropDown(): void {
-		const dropdownBtn = document.querySelector('#dropdown_btn') as HTMLElement;
-		const dropdown = document.querySelector('.dropdown') as HTMLInputElement;
-
+		const dropdownBtn = document.querySelector('#dropdown-btn') as HTMLElement;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts]: Using ids with very generic names like `dropdown-btn` and `dropdown` seems like a recipe for disaster if you're querying the whole document. I'd suggest scoping down the search (which is a good idea anyways to reduce the overhead of having to traverse the entire DOM which shouldn't be necessary)

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts]: Same goes for a lot of the other queries you're doing.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 3]
```diff
@@ -968,6 +1005,7 @@ registerThemingParticipant((theme, collector) => {
 	const tileBackgroundColor = theme.getColor(inputBackground);
 	if (tileBackgroundColor) {
 		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer .tile:not(.extension):not(.extension-pack) { background-color: ${tileBackgroundColor};  }`);
+		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer .btn-secondary .monaco-button { background-color: ${tileBackgroundColor} !important;  }`);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:1009]: Why is the !important needed here? Is there something else setting the color you need to override?

Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:1009]: Hey Charles, yeah i didnt want to do that. I was hoping to get some feedback. This is to override the inline color that are set on the monaco button. Any suggestions?

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 4]
```diff
@@ -334,11 +337,75 @@ class WelcomePage extends Disposable {
...
 			}
 		}));
+		this.createButtons();
 		this.createDropDown();
 		this.createWidePreviewToolTip();
 		this.createPreviewModal();
 	}
 
+	private createButtons(): void {
+		const dropdownButtonContainer = document.querySelector('#dropdown-btn-container') as HTMLElement;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:347]: Using ids with very generic names like dropdown-btn-container seems like a recipe for disaster if you're querying the whole document. I'd suggest scoping down the search (which is a good idea anyways to reduce the overhead of having to traverse the entire DOM which shouldn't be necessary)

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:1009]: I'd suggest looking into calling button.style directly - e.g. https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/browser/ui/dialog/dialog.ts#L295

**Pull request merged**